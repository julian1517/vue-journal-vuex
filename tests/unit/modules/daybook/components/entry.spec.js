
import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";






describe('Pruebas en el Entry Component', () => {

    const mockRouter= {
        push: jest.fn()
    }

    const wrapper = shallowMount( Entry, {
        
        props: {
            entry: {   
                id:'-NRAL-bBmcjhl9CIUFGd',
                date: 1679522515719,
                text: "hola mundo desde mock data"
        },
                
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }

    })


    test('debe de hacer match con el snapshot', () => {
        
        
        expect( wrapper.html()).toMatchSnapshot()
    
    })

    test('debe de redireccionar al hacer click en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')
        
        expect( mockRouter.push).toHaveBeenCalledWith(
            {
                name:'entry',
                params: {
                    id:'-NRAL-bBmcjhl9CIUFGd'
                }

            }
        )
    
    })

    test('pruebas en las propiedades computadas', () => {
        
        
        expect(wrapper.vm.day).toBe(22)
        expect(wrapper.vm.month).toBe('Marzo')
        expect(wrapper.vm.yearDay).toBe('2023,Mi√©rcoles')
        
    
    })
})