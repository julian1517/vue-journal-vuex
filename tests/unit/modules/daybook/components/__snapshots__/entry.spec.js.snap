// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">22</span><span class="mx-1 fs-5">Marzo</span><span class="mx-2 fw-light">2023,Mi√©rcoles]</span></div>
  <div class="entry-description">hola mundo desde mock data</div>
</div>
`;
